packages/suins/sources/pricing.move

```diff
@@ -0,0 +1,78 @@
+module suins::pricing;
+
+use sui::vec_map::{Self, VecMap};
+
+// Tries to create a range with more than 2 values
+const EInvalidLength: u64 = 1;
+/// Tries to create a range with the first value greater than the second
+const EInvalidRange: u64 = 2;
+/// Tries to create a pricing config with different lengths for ranges and
+/// prices
+const ELengthMissmatch: u64 = 3;
+/// Tries to calculate the price for a given length
+const EPriceNotSet: u64 = 4;
+
+/// A range struct that holds the start and end of a range (inclusive).
+public struct Range(u64, u64) has copy, store, drop;
+
+/// A struct that holds the length range and the price of a service.
+public struct PricingConfig has copy, store, drop {
+    pricing: VecMap<Range, u64>,
+}
+
+public struct RenewalConfig has copy, store, drop {
+    config: PricingConfig,
+}
+
+/// Calculates the price for a given length.
+/// Aborts with EPriceNotSet if the price for the given length is not set.
+public fun calculate_price(config: &PricingConfig, length: u64): u64 {
```

Comment:
Need to update logic here after oracle integration, perhaps calculate_price<T> and then conversion from the base to the input token, also check input token is in allowlist (SUI, NS, USDC as example)